### PR TITLE
Replace option_try macros and match with ? operator

### PR DESCRIPTION
**Pull request by mbrubeck**

### CODE REVIEW

**Comment by kennytm** [comment]
Could you also replace:

* `try_opt!` macro in `src/libsyntax_ext/format_foreign.rs` and `src/libstd/net/addr.rs`, 
* The hand-written expansions in `src/liballoc/allocator.rs`
* Everywhere-else appropriate where `None => return None` appears

Thanks!

**Comment by mbrubeck** [comment]
Updated to replace `try_opt!` and various manual `match` expressions.  I'm sure there are more places where this could be used, but this covers the ones that were easily findable and relatively straightforward to replace.

File: src/librustdoc/html/format.rs [hunk 1]
```diff
@@ -425,15 +425,13 @@ pub fn href(did: DefId) -> Option<(String, ItemType, Vec<String>)> {
...
-            Some(&(ref fqp, shortty)) => {
-                (fqp, shortty, match cache.extern_locations[&did.krate] {
-                    (.., render::Remote(ref s)) => s.to_string(),
-                    (.., render::Local) => repeat("../").take(loc.len()).collect(),
-                    (.., render::Unknown) => return None,
-                })
-            }
-            None => return None,
+        None => {
+            let &(ref fqp, shortty)) = cache.external_paths.get(&did)?;
```
Comment (kennytm) [inline src/librustdoc/html/format.rs]: There’s an extra `)` here

**Review by kennytm — APPROVED** [review]
r=me after Travis is passing.

File: src/librustdoc/html/format.rs (hunk 1 above)
Comment (mbrubeck) [reply src/librustdoc/html/format.rs]: Fixed.

**Comment by leonardo-m** [comment]
A nice reduction of lines count.

**Pull request merged**